test/test_cuda.py

```diff
@@ -2450,7 +2450,12 @@ def _grad_scaling_autocast_fused_optimizers(self, optimizer_ctor, optimizer_kwar
                     actual = state_scaling[k]
                     if k == "step":
                         actual = actual.squeeze()
-                    self.assertEqual(state_control[k], actual, msg=k)
+
```

Comment:
Why does this need to be updated? Would passing `exact_dtype=False` to `assertEqual` be enough?

Reply:
Actually removed this once I put the cast into `step`